stdlib/public/core/StringBridge.swift

```diff
@@ -282,30 +299,36 @@ private var kCFStringEncodingUTF8: _swift_shims_CFStringEncoding {
 }
 
 internal enum _KnownCocoaString {
-  case storage
-  case shared
-  case cocoa
+  case storage(__StringStorage)
+  case shared(__SharedStringStorage)
+  case mutable(_SwiftNSMutableString)
+  case cocoa(_CocoaString)
 #if !(arch(i386) || arch(arm))
-  case tagged
+  case tagged(_CocoaString)
```

Comment:
I seem to recall moving to a trivial enum because sinking the object into an associated value caused extra ARC traffic. Is this no longer the case?

Reply:
oh interesting. Now that you bring it up I vaguely remember that too. This is back to perf neutral at this point so apparently it's ok, but maybe I accidentally got enough wins to counter out this part regressing it and ended up neutral overall.

Reply:
Worth a check